erpnext/projects/doctype/task/task.js

```diff
@@ -66,13 +66,13 @@ frappe.ui.form.on("Task", {
 	},
 
 	validate: function (frm) {
-		frm.doc.project && frappe.model.remove_from_locals("Project",
-			frm.doc.project);
+		frm.doc.default_project && frappe.model.remove_from_locals("Default_Project",
+			frm.doc.default_project);
 	},
 
-	project: (frm) => {
-		if (frm.doc.project && frm.doc.billable === 0) {
-			frappe.db.get_value("Project", { "name": frm.doc.project }, "billable", (r) => {
+	default_project: (frm) => {
```

Comment:
will also have to add a JS trigger if a project is set as default on child table to ensure that this block of code runs